File: server-side2/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SQLink.Models
{
    public class Transaction
    {
        [Required]
        [MinLength(1)]
        [MaxLength(64)]
        public string Id { get; set; } = string.Empty;

        [Required]
        [Range(0.01, double.MaxValue)]
        public decimal Amount { get; set; }

        [Required]
        [StringLength(3, MinimumLength = 3)]
        [RegularExpression("^[A-Z]{3}$")]
        public string Currency { get; set; } = "USD";

        [Required]
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;

        [Required]
        [RegularExpression("^(Pending|Completed|Failed)$")]
        public string Status { get; set; } = "Pending";
    }
}

File: SQLINK-home-assignment-frontend/src/types/transaction.ts
export type TransactionStatus = "Pending" | "Completed" | "Failed";

export interface Transaction {
  id: string;
  amount: number;
  currency: string;
  timestamp: string;
  status: TransactionStatus;
}

export interface IngestResponse {
  accepted: boolean;
  id?: string;
  message?: string;
}

export interface TransactionFilter {
  showOnlyErrors: boolean;
}
